source/slang/slang-diagnostic-defs.h

```diff
@@ -245,6 +245,10 @@ DIAGNOSTIC(29100, Error, unrecognizedSPIRVOpcode, "unrecognized spirv opcode: $0
 DIAGNOSTIC(29101, Error, misplacedResultIdMarker, "the result-id marker must only be used in the last instruction of a spriv_asm expression")
 DIAGNOSTIC(29102, Note,  considerOpCopyObject, "consider adding an OpCopyObject instruction to the end of the spirv_asm expression")
 DIAGNOSTIC(29103, Note,  noSuchAddress, "unable to take the address of this address-of asm operand")
+DIAGNOSTIC(29104, Error, spirvInstructionWithoutResultId, "cannot use this 'x = $0...' syntax because $0 does not have a <result-id> operand")
+DIAGNOSTIC(29104, Error, spirvInstructionWithoutResultTypeId, "cannot use this 'x : <type> = $0...' syntax because $0 does not have a <result-type-id> operand")
+// This is a warning because we trust that people using the spirv_asm block know what they're doing
+DIAGNOSTIC(29104, Warning, spirvInstructionWithTooManyOperands, "Too many operands for $0 (expected max $1), did you forget a semicolon?")
```

Comment:
If we see a spvword that starts with `Op***`, we can tell it is a new inst, right?

Reply:
Sadly not, they can be used for example here: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpSpecConstantOp

Reply:
Oh, that's too bad...